### PR TITLE
Adding auth pages

### PR DESCRIPTION
**Pull request by b0llu**
## What to check
- [ ] src/Components/PrivateRoute/RequireAuth.jsx
- [ ] src/Components/PrivateRoute/RestrictAuth.jsx
- [ ] src/Context/Auth.context.js
- [ ] src/Context/Reducer.context.js
- [ ] src/Reducer/reducer.js
- [ ] src/App.js

## What not to check
- [ ] Do Not Check Any Other Files

image
image

### CODE REVIEW

File: src/Pages/AuthPage/ProfileComponents/SignupBox/SignupBox.jsx [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    passwordOne: "",
+    passwordTwo: "",
+    terms: false,
+  });
+  const [error, setError] = useState({ state: false, text: "" });
+  const [type, setType] = useState(true);
+
+  const validation = /^(?=.*\d)(?=.*[a-z]).{5,10}$/;
+  const emailValidation =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```
Comment (Developers-codz) [inline src/Pages/AuthPage/ProfileComponents/SignupBox/SignupBox.jsx:21]: Its awesome that you used regex for email validation

**Review by Developers-codz — APPROVED** [review]
Everything looks good. Go ahead

File: src/Pages/AuthPage/ProfileComponents/SignupBox/SignupBox.jsx (hunk 1 above)
Comment (b0llu) [reply src/Pages/AuthPage/ProfileComponents/SignupBox/SignupBox.jsx:21]: Thanks!

File: src/Context/Auth.context.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const [userState, setUserState] = useState([]);
+
+  const login = async (userDetails) => {
+    try {
+      const { data } = await axios.post(`/api/auth/login`, {
+        email: userDetails.email,
+        password: userDetails.password,
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("StormKeepToken", data.encodedToken);
```
Comment (arjun-computer-geek) [inline src/Context/Auth.context.js:19]: This is great way of doing code that write comments what are you doing. Great👍

File: src/Context/Auth.context.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        firstName: userDetails.user,
+        email: userDetails.email,
+        password: userDetails.passwordOne,
+      });
+      // saving the encodedToken in the localStorage
+      localStorage.setItem("StormKeepToken", data.encodedToken);
+      localStorage.setItem("StormKeepUser", data.createdUser.firstName);
+      dispatch({ type: "SUCCESS_TOAST", payload: "Sign Up Successful" });
+    } catch (error) {
+      console.log(error);
```
Comment (arjun-computer-geek) [inline src/Context/Auth.context.js:39]: I think it can also shown on toast? Right?

**Review by arjun-computer-geek — APPROVED** [review]
All looks good. It's very hard to find any bug or mistake. Cleaner code. Good job👍

File: src/Context/Auth.context.js (hunk 3 above)
Comment (b0llu) [reply src/Context/Auth.context.js:39]: Thanks for pointing it out i actually forgot to show error with a toast here

**Pull request merged**